pkg/controllers/management/rkeworkerupgrader/upgrade.go

```diff
@@ -369,6 +384,45 @@ func (uh *upgradeHandler) toUpgradeCluster(cluster *v3.Cluster) (bool, error) {
 	return false, nil
 }
 
+func (uh *upgradeHandler) restore(cluster *v3.Cluster) error {
+	nodes, err := uh.nodeLister.List(cluster.Name, labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	// allowed to use goroutine?
+	var errgrp errgroup.Group
+	nodesQueue := util.GetObjectQueue(nodes)
+	for w := 0; w < 5; w++ {
```

Comment:
What's the reason to use goroutinue here? Do you need to execute the logic fives times and wait for all go routinues be to finished?

Reply:
each goroutine reads one node from the queue as and when it's done. 
ProcessNode gets the plan and updates the node so I didn't want to do that in a normal for loop, but also spinning one goroutine for each node seemed like a bit too much. So I am creating 5 goroutines and wait till entire queue is done